core/lib/dal/src/blocks_web3_dal.rs

```diff
@@ -71,6 +69,15 @@ impl BlocksWeb3Dal<'_, '_> {
 
         let block = rows.into_iter().fold(None, |prev_block, row| {
             let mut block = prev_block.unwrap_or_else(|| {
+                let (l1_batch_number, l1_batch_timestamp) =
+                    if row.l1_batch_is_sealed.unwrap_or(false) {
```

Comment:
nit: you can do `LEFT JOIN l1_batches ON l1_batches.number = miniblocks.l1_batch_number AND l1_batches.is_sealed = TRUE` instead of selecting `l1_batch_is_sealed`

Reply:
I need l1_batch_is_sealed for returning pending batches in a way it was in api before

Reply:
but adding `l1_batches.is_sealed = TRUE` to the JOIN will do the job, no?

Reply:
hm, from the second glance yes. let's see in tests